packages/wp-checkout/src/components/checkout.js

```diff
@@ -160,63 +156,46 @@ function BillingDetailsStep( { isActive, isComplete, setStepNumber, onChangeBill
 	const localize = useLocalize();
 	const [ paymentData, setPaymentData ] = usePaymentMethodData();
 	const paymentMethod = usePaymentMethod();
-	if ( ! paymentMethod ) {
```

Comment:
It's unlikely, but possible that this component might be rendered when `usePaymentMethod()` returns null, which would cause a fatal error here if we remove this block. However, since this is an internal component and we can pretty much guarantee it won't be null (we select the first payment method by default), we could change this block to just throw an error.

```
if ( ! paymentMethod ) {
  throw new Error( 'Cannot render Billing details without a payment method' );
}
```

Reply:
That's good point. This will most likely only happen in our dev environments though right? I wonder if it's necessary?

Reply:
IMO it's not worth leaving potential failures unguarded. A thrown Error will at give us a clear message which can be used to help track down a bug. Otherwise the user or developer would see something like `Cannot read property 'BillingDetailsComponent' of undefined` which is much less actionable than `Cannot render Billing details without a payment method`.